hack/code/prices_gen.go

```diff
@@ -39,12 +39,50 @@ import (
 	"github.com/aws/karpenter/pkg/test"
 )
 
-func main() {
+func getAWSRegions(partition string) []string {
+	switch partition {
+	case "aws":
+		return []string{"us-east-1"}
+	case "aws-us-gov":
+		return []string{"us-gov-east-1", "us-gov-west-1"}
+	case "aws-cn":
+		return []string{"cn-north-1"}
+	default:
+		panic("invalid partition")
+	}
+}
+
+func getPartitionSuffix(partition string) string {
+	switch partition {
+	case "aws":
+		return "AWS"
+	case "aws-us-gov":
+		return "USGov"
+	case "aws-cn":
+		return "CN"
+	default:
+		panic("invalid partition")
+	}
+}
+
+type Options struct {
+	partition string
+	output    string
+}
+
+func NewOptions() *Options {
+	o := &Options{}
+	flag.StringVar(&o.partition, "partition", "aws", "The partition to generate prices for. Valid options are \"aws\", \"aws-us-gov\", and \"aws-cn\".")
+	flag.StringVar(&o.output, "output", "pkg/providers/pricing/zz_generated.pricing_us.go", "The destination for the generated go file.")
 	flag.Parse()
-	if flag.NArg() != 1 {
-		log.Fatalf("Usage: %s pkg/providers/pricing/zz_generated.pricing.go", os.Args[0])
+	if !lo.Contains([]string{"aws", "aws-us-gov", "aws-cn"}, o.partition) {
+		panic("invalid partition: must be \"aws\", \"aws-us-gov\", or \"aws-cn\"")
```

Comment:
```suggestion
		log.Fatal("invalid partition: must be \"aws\", \"aws-us-gov\", or \"aws-cn\"")
```

nit: prefer `log.Fatal` over `panic` for consistency and convention